subsys/bluetooth/host/scan.c

```diff
@@ -1588,7 +1588,11 @@ int bt_le_scan_stop(void)
 
 void bt_le_scan_cb_register(struct bt_le_scan_cb *cb)
 {
-	sys_slist_append(&scan_cbs, &cb->node);
+	sys_snode_t *prev_node;
+
+	if (!sys_slist_find(&scan_cbs, &cb->node, &prev_node)) {
+		sys_slist_append(&scan_cbs, &cb->node);
+	}
```

Comment:
The documentation for the function doesn't seem to very explicit about it, but it looks like you can pass `NULL` to the prev_node
```suggestion
	if (!sys_slist_find(&scan_cbs, &cb->node, NULL)) {
		sys_slist_append(&scan_cbs, &cb->node);
	}
```